### PR TITLE
Fix tests on TPU

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -622,7 +622,8 @@ at::Tensor AtenXlaType::mm(const at::Tensor& self,
 at::Tensor AtenXlaType::matmul(const at::Tensor& self,
                                const at::Tensor& other) const {
   return bridge::AtenFromXlaTensor(XLATensor::matmul(
-      bridge::GetXlaTensor(self), bridge::GetXlaTensor(other)));
+      bridge::GetXlaTensor(self), bridge::GetXlaTensor(other),
+      /*use_full_conv_precision=*/s_use_full_conv_precision_));
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: No please, let's get rid of this.
It is a global flag anyway, and there is no reason to poison all the APIs from up the call chain.
Let's have a global (like currently have) flag, maybe initialized by environment as well, and leave the APIs alone.
Only when we call the lower level Dot/Conv/DotGeneral/... we will ask for the global flag value.
Stick it in helpers.cpp/h

Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

File: torch_xla/csrc/helpers.cpp [hunk 2]
```diff
@@ -11,6 +11,8 @@
 
 namespace torch_xla {
 
+bool XlaHelpers::s_use_full_mat_mul_precision = false;
```
Comment (dlibenzi) [inline torch_xla/csrc/helpers.cpp]: Maybe a set/get access was better, but no need to do it now.

File: torch_xla/csrc/ops/ops.cpp [hunk 3]
```diff
@@ -199,16 +199,20 @@ NodePtr Dot(const Value& input, const Value& weight,
                             output_shape, std::move(lower_fn));
 }
 
-NodePtr MatMul(const Value& lhs, const Value& rhs) {
-  auto lower_fn = [](const ir::Node& node,
-                     ir::LoweringContext* loctx) -> ir::XlaOpVector {
+NodePtr MatMul(const Value& lhs, const Value& rhs,
+               bool use_full_conv_precision) {
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/ops.cpp]: Push it deeper. At xla:: call time.

File: torch_xla/csrc/ops/ops.h [hunk 4]
```diff
@@ -106,7 +106,8 @@ NodePtr AddMatMulOp(const Value& input, const Value& weight, const Value& bias,
 NodePtr Dot(const Value& input, const Value& weight,
             bool use_full_conv_precision);
 
-NodePtr MatMul(const Value& lhs, const Value& rhs);
+NodePtr MatMul(const Value& lhs, const Value& rhs,
+               bool use_full_conv_precision);
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/ops.h]: We should not have this flag in the APIs at all.

Comment (asuhan) [reply torch_xla/csrc/ops/ops.h]: So now we're removing this from everywhere, including nodes? I distinctly remember the discussion when we first added it, that a global flag would be bad. But ok.

Comment (dlibenzi) [reply torch_xla/csrc/ops/ops.h]: We just use them when we call XLA.

xla::Dot(x, y, gimme_the_flag());

That is by any means a global thing, are by polluting the APIs we are trying to make it look like a non global, but its value is indeed global.
Better leave the APIs clean. Nobody cares besides tests.
Given that you do the changes, can you use a get/set API for it?

Comment (asuhan) [reply torch_xla/csrc/ops/ops.h]: Note that I now have to change it *everywhere*, including the `XlaModule`. Do you still think this is a good idea?

Comment (dlibenzi) [reply torch_xla/csrc/ops/ops.h]: I love it! :)
Mucho bonus points!

File: torch_xla/csrc/helpers.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/helpers.cpp]: Done.

File: torch_xla/csrc/ops/ops.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/ops/ops.cpp]: Done.

File: torch_xla/csrc/ops/ops.h (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/ops/ops.h]: Done.

**Pull request merged**